File: src/main/java/schemabuilder/processor/pipelines/parsing/scalars/GraphQLScalarBank.java
package schemabuilder.processor.pipelines.parsing.scalars;

import java.util.HashSet;
import java.util.Set;

public class GraphQLScalarBank {

    private static Set<GraphQLScalarType> scalars = new HashSet<>();


    private static GraphQLScalarBank instance = null;

    public static GraphQLScalarBank getInstance() {
        if (instance == null) {
            instance = new GraphQLScalarBank();
        }
        return instance;
    }

    private GraphQLScalarBank() {}


    public void addScalar(GraphQLScalarType dataFetcher) {
        scalars.add(dataFetcher);
    }

    public Set<GraphQLScalarType> getScalars() {
        return scalars;
    }
}

File: src/main/java/schemabuilder/processor/pipelines/parsing/scalars/GraphQLScalarBank.kt
package schemabuilder.processor.pipelines.parsing.scalars

object GraphQLScalarBank {
    fun addScalar(dataFetcher: GraphQLScalarType?) {
        scalars.add(dataFetcher)
    }

    val scalars: MutableSet<GraphQLScalarType?> = mutableSetOf()
}